frontend/javascripts/maintenance_banner.tsx

```diff
@@ -0,0 +1,106 @@
+import {
+  MaintenanceInfo,
+  listCurrentAndUpcomingMaintenances,
+  updateNovelUserExperienceInfos,
+} from "admin/admin_rest_api";
+import { Alert } from "antd";
+import FormattedDate from "components/formatted_date";
+import { useInterval } from "libs/react_helpers";
+import { setActiveUserAction } from "oxalis/model/actions/user_actions";
+import { Store } from "oxalis/singletons";
+import React, { useState } from "react";
+
+export function MaintenanceBanner() {
+  const { activeUser } = Store.getState();
+  const [currentAndUpcomingMaintenances, setcurrentAndUpcomingMaintenances] = useState<
+    Array<MaintenanceInfo>
+  >([]);
+  const [position, setPosition] = useState<Object>({ top: 48 });
+  const [isTop, setIsTop] = useState(true);
+  useInterval(
+    async () => {
+      if (activeUser == null) return;
+      setcurrentAndUpcomingMaintenances(await listCurrentAndUpcomingMaintenances());
+    },
+    10000, //TODO find value for production
+  );
+  if (activeUser == null) return <></>;
+  const activeUsersLatestAcknowledgedMaintenance =
+    activeUser?.novelUserExperienceInfos.latestAcknowledgedMaintenanceInfo;
+
+  const saveUserClosedMaintenanceInfo = (closestUpcomingMaintenance: MaintenanceInfo) => {
+    if (activeUser == null) return;
+    const [nextMaintenanceAcknowledged] = updateNovelUserExperienceInfos(activeUser, {
+      latestAcknowledgedMaintenanceInfo: closestUpcomingMaintenance.id,
+    });
+    Store.dispatch(setActiveUserAction(nextMaintenanceAcknowledged));
+  };
+
+  const toggleTopOrBottomPosition = () => {
+    setPosition(isTop ? { top: "48px" } : { bottom: "20px" });
+    setIsTop(!isTop);
+  };
+
+  const getClosestUpcomingMaintenanceBanner = () => {
+    const currentTime = Date.now();
+    const closestUpcomingMaintenance = currentAndUpcomingMaintenances
+      ?.filter((maintenance) => maintenance.startTime > currentTime)
+      .sort((a, b) => a.startTime - b.startTime)[0];
```

Comment:
isn't this redundant to the rest admin module? at least the sorting. I understand, that a running maintenance can become finished while this component is updating.

Reply:
my thought was that I didn't want to depend on the rest admin module returning a sorted array (maybe a refactoring might change this), so I'll just do it again. But in the admin rest API I thought that it would be nice to return the maintenances sorted. Then I think I'll just remove the sorting from the admin rest module